internal/resources/aws/opensearch_domain.go

```diff
@@ -0,0 +1,219 @@
+package aws
```

Comment:
Unless I'm mistaken this file is identical to the `elasticsearch_domain` resource. I'd prefer that we didn't just duplicate this and instead just use the `ElasticsearchDomain` resource. Perhaps we can rename `ElasticsearchDomain` it to `SearchDomain` and add doc blocks to convey that it is used for both `OpensearchDomain` and `ElasticsearchDomain`.